sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -84,15 +84,21 @@ class SessionCatalog(
   // ----------------------------------------------------------------------------
 
   def createDatabase(dbDefinition: CatalogDatabase, ignoreIfExists: Boolean): Unit = {
-    externalCatalog.createDatabase(dbDefinition, ignoreIfExists)
+    if (!databaseExists(dbDefinition.name)) {
+      externalCatalog.createDatabase(dbDefinition, ignoreIfExists)
+    }
```

Comment:
For createDatabase, we should throw an exception if db already exists and ignoreIfExists is false, right?